### PR TITLE
BAH-1147 | BAH-1148

### PR DESCRIPTION
**Pull request by rupamJoshi**
BAH-1147 | Null is Getting printed in sample lable print for collection date. -> https://bahmni.atlassian.net/browse/BAH-1147

BAH-1148 | File uploaded while entering the results for referred out tests is not saved until and unless the "Mark as Done" flag is checked before saving it. -> https://bahmni.atlassian.net/browse/BAH-1148

### CODE REVIEW

File: openelis/WebContent/pages/dashboard/dashboardMaster.jsp [hunk 1]
```diff
@@ -372,9 +372,12 @@ basePath = path + "/";
...
             onSuccess:  function onLabelSelected(xhr) {
                 var datePattern = '<%=SystemConfiguration.getInstance().getPatternForDateLocale() %>';
-                var collectionDate=collectionDateTime.slice(0,10);
-                var collectionTime=OpenElis.Utils.getTime(collectionDateTime.slice(11,19));
-                collectionDateTime=collectionDate+" "+collectionTime;
+                var locale = '<%=SystemConfiguration.getInstance().getDateLocale() %>';
+                if(collectionDateTime!=null) {
+                    var collectionDate = new Date(collectionDateTime).toLocaleDateString(locale.replace("_","-"));
+                }else {
+                    collectionDate="";
```
Comment (angshu) [inline openelis/WebContent/pages/dashboard/dashboardMaster.jsp:379]: should this variable be declared outside of the if clause?

File: openelis/src/us/mn/state/health/lims/common/provider/query/TestUpdateWithAccessionNumberProvider.java [hunk 2]
```diff
@@ -44,6 +45,11 @@ public void processRequest(HttpServletRequest request, HttpServletResponse respo
         String sampleId = request.getParameter("sampleId");
 
         Sample sample = sampleDao.getSampleByID(sampleId);
+        String datePattern = ResourceLocator.getInstance().getMessageResources().getMessage( "date.format.formatKey");
```
Comment (angshu) [inline openelis/src/us/mn/state/health/lims/common/provider/query/TestUpdateWithAccessionNumberProvider.java:48]: is this property already present and set?

File: openelis/WebContent/pages/dashboard/dashboardMaster.jsp [hunk 3]
```diff
@@ -372,9 +372,12 @@ basePath = path + "/";
             parameters: "provider=PatientSearchPopulateProvider&stNumber=" + stNumber,
             onSuccess:  function onLabelSelected(xhr) {
                 var datePattern = '<%=SystemConfiguration.getInstance().getPatternForDateLocale() %>';
-                var collectionDate=collectionDateTime.slice(0,10);
-                var collectionTime=OpenElis.Utils.getTime(collectionDateTime.slice(11,19));
-                collectionDateTime=collectionDate+" "+collectionTime;
+                var locale = '<%=SystemConfiguration.getInstance().getDateLocale() %>';
+                if(collectionDateTime!=null) {
+                    var collectionDate = new Date(collectionDateTime).toLocaleDateString(locale.replace("_","-"));
```
Comment (angshu) [inline openelis/WebContent/pages/dashboard/dashboardMaster.jsp:377]: not sure why we have to locale.replace? also, we are reading the locale of the backend server system, and using that to show "locale" specific date on the frontend?

File: openelis/src/us/mn/state/health/lims/common/provider/query/TestUpdateWithAccessionNumberProvider.java (hunk 2 above)
Comment (rupamJoshi) [reply openelis/src/us/mn/state/health/lims/common/provider/query/TestUpdateWithAccessionNumberProvider.java:48]: yes, it was already present and set in MessageResources.properties, and this piece of code was written to fix the null collection date issue.

Comment (rupamJoshi) [reply openelis/src/us/mn/state/health/lims/common/provider/query/TestUpdateWithAccessionNumberProvider.java:48]: Now Code is changes and we using different approach.

File: openelis/WebContent/pages/dashboard/dashboardMaster.jsp (hunk 1 above)
Comment (rupamJoshi) [reply openelis/WebContent/pages/dashboard/dashboardMaster.jsp]: Now Code is changes and we using different approach.

**Pull request merged**